packages/contracts-bedrock/scripts/DeploySuperchain.s.sol

```diff
@@ -0,0 +1,383 @@
+ * inadvertently transpose two addresses, for example).
+ *
+ * Each deployment step in the core deploy script is modularized into its own function that performs
+ * the deploy and sets the output on the Output contract, allowing for easy composition and testing
+ * of deployment steps. The output setter methods requires keying off the four-byte selector of the
+ * each output field's getter method, ensuring that the output is set for the correct field and
+ * minimizing the amount of boilerplate needed for each output field.
+ *
+ * This script doubles as a reference for documenting the pattern used and therefore contains
+ * comments explaining the patterns used. Other scripts are not expected to have this level of
+ * documentation.
+ *
+ * Additionally, we intentionally use "Input" and "Output" terminology to clearly distinguish these
+ * scripts from the existing ones that "Config" and "Artifacts" terminology.
+ */
+contract DeploySuperchainInput {
+    // The input struct contains all the input data required for the deployment.
+    struct Input {
+        Roles roles;
+        bool paused;
+        ProtocolVersion requiredProtocolVersion;
+        ProtocolVersion recommendedProtocolVersion;
+    }
+
+    struct Roles {
+        address proxyAdminOwner;
+        address protocolVersionsOwner;
+        address guardian;
+    }
+
+    // This flag tells us if all inputs have been set. An `input()` getter method that returns all
+    // inputs reverts if this flag is false. This ensures the deploy script cannot proceed until all
+    // inputs are validated and set.
+    bool public inputSet = false;
+
+    // The full input struct is kept in storage. It is not exposed because the return type would be
+    // a tuple, but it's more convenient for the return type to be the struct itself. Therefore the
+    // struct is exposed via the `input()` getter method.
+    Input internal inputs;
+
+    // And each field is exposed via it's own getter method. We can equivalently remove these
+    // storage variables and add getter methods that return the input struct fields directly, but
+    // that is more verbose with more boilerplate, especially for larger scripts with many inputs.
+    // Unlike the `input()` getter, these getters do not revert if the input is not set. The caller
+    // should check the `inputSet` value before calling any of these getters.
+    address public proxyAdminOwner;
+    address public protocolVersionsOwner;
+    address public guardian;
+    bool public paused;
+    ProtocolVersion public requiredProtocolVersion;
+    ProtocolVersion public recommendedProtocolVersion;
+
+    // Load the input from a TOML file.
+    function loadInputFile(string memory _infile) public {
+        _infile;
+        Input memory parsedInput;
+        loadInput(parsedInput);
+        require(false, "DeploySuperchainInput: loadInput is not implemented");
```

Comment:
Did you mean to remove this `require` statement? It looks like its implememented ?

Reply:
This method will parse the input TOML file into the `Input` struct, but that file parsing is not currently implemented